types/src/on_chain_config/execution_config.rs

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::on_chain_config::OnChainConfig;
+use anyhow::{format_err, Result};
+use serde::{Deserialize, Serialize};
+
+/// The on-chain execution config, in order to be able to add fields, we use enum to wrap the actual struct.
+#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Serialize)]
+pub enum OnChainExecutionConfig {
```

Comment:
@zekun000 - This needs to be an on-chain config as this needs to be coordinated across validated at the same time. Otherwise, the execution result across validators will be different and will cause a network stall :)